tests/test_rechunking.py

```diff
@@ -1,14 +1,117 @@
+import itertools
 import random
+from collections import namedtuple
 
+import numpy as np
 import pytest
 import xarray as xr
 
-from pangeo_forge_recipes.rechunking import combine_fragments, split_fragment
+from pangeo_forge_recipes.rechunking import GroupKey, combine_fragments, split_fragment
 from pangeo_forge_recipes.types import CombineOp, Dimension, Index, IndexedPosition, Position
 
+from .conftest import split_up_files_by_variable_and_day
 from .data_generation import make_ds
 
 
+@pytest.mark.parametrize(
+    "nt_dayparam",
+    [(5, "1D"), (10, "2D")],
+)
+@pytest.mark.parametrize("time_chunks", [1, 2, 5])
+def test_split_and_combine_fragments_with_merge_dim(nt_dayparam, time_chunks):
+    """Test if sub-fragments split from datasets with merge dims can be combined with each other."""
+
+    target_chunks = {"time": time_chunks}
+    nt, dayparam = nt_dayparam
+    ds = make_ds(nt=nt)
+    dsets, _, _ = split_up_files_by_variable_and_day(ds, dayparam)
+
+    # replicates indexes created by IndexItems transform.
+    unique_times = np.unique([ds.time[0].values for ds in dsets])
+    time_positions = {t: i for i, t in enumerate(unique_times)}
```

Comment:
This logic seems wrong. The index in `IndexedPosition` needs to be an actual offset from the beginning of the array.